tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/BlobUploadProcessor.cs

```diff
@@ -286,5 +295,101 @@ await using (var messagesWriter = new StreamWriter(File.OpenWrite(tempFile)))
                 }
             }
         }
+
+        private async Task UploadTestRunBlobsAsync(string account, Build build)
+        {
+            try
+            {
+                var continuationToken = string.Empty;
+                var buildIds = new[] { build.Id };
+                var minLastUpdatedDate = build.QueueTime.Value.AddHours(-1);
+                var maxLastUpdatedDate = build.FinishTime.Value.AddHours(1);
+
+                do
+                {
+                    var page = await testResultsClient.QueryTestRunsAsync2(
+                        build.Project.Id,
+                        minLastUpdatedDate,
+                        maxLastUpdatedDate,
+                        continuationToken: continuationToken,
+                        buildIds: buildIds
+                    );
+
+                    foreach (var testRun in page)
+                    {
+                        await UploadTestRunBlobAsync(account, build, testRun);
```

Comment:
Is it worth using something like `WhenAll` here or is it always a pretty low number/fast upload?

Reply:
It's pretty low count / fast normally.   If it gets large, I'd want to use some concurrency limiting parallel async.  maybe something as simple as `foreach(var testRun in page.AsParallel())`

Reply:
Because the list is typically small, I just foreach.  If it did get large, I'd want to use concurrency limiting parallel tasks to avoid starting 100 blob uploads at once. That may be a simple as Task.WhenAll + SemaphoreSlim